### PR TITLE
[chrony] Adding a dot to the end of each NTP server address to avoid resolve problems. Allowing correction of large time difference

### PR DESCRIPTION
**Pull request by duckhawk**
## Description

Adding a dot to the end of each NTP server address to avoid resolve problems. Allowing correction of large time difference.

## Why do we need it, and what problem does it solve?

Incorrect DNS resolve is possible in case of wildcard domains in search. In case of a large difference in local time with the world time correction may not work. This PR corrects both situations.

## What is the expected result?

Chrony correct work.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: chrony
type: fix
summary: Adding a dot to the end of each NTP server address to avoid resolve problems. Allowing correction of large time difference.
impact_level: low
```

### CODE REVIEW

**Review by z9r5 — DISMISSED** [review]

File: modules/470-chrony/images/chrony/start.sh [hunk 1]
```diff
@@ -34,10 +34,13 @@ cmdallow 127/8
...
 bindaddress 127.0.0.1
 driftfile /var/run/chrony/chrony.drift
-makestep 1.0 3
+makestep 1.0 -1
 rtcsync
 EOF
 for NTP_SERVER in ${NTP_SERVERS}; do
+  if [ ${NTP_SERVER:0-1} != "." ]; then
+    NTP_SERVER=$NTP_SERVER"."
+  fi
```
Comment (RomanenkoDenys) [inline modules/470-chrony/images/chrony/start.sh:43]: Please use trimSuffix func in helm for it.

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]

File: modules/470-chrony/images/chrony/start.sh (hunk 1 above)
Comment (duckhawk) [reply modules/470-chrony/images/chrony/start.sh:43]: But we need to add a dot here as a suffix, not to remove it.

Comment (duckhawk) [reply modules/470-chrony/images/chrony/start.sh]: This script runs on container startup, so we can't use helm functions here. Also, I'd added check for IP address in ntp servers.

File: modules/470-chrony/images/chrony/start.sh [hunk 2]
```diff
@@ -34,11 +34,20 @@ cmdallow 127/8
...
+  # Check, if there is an IP address, then pass it to file
+  if [[ $NTP_SERVER =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+      echo "pool ${NTP_SERVER} iburst" >> /var/run/chrony/chrony.conf
+  # Else, if there is a dot in the end of DNS-address, then pass it to file
+  elif [[ $NTP_SERVER =~ ^.*\.$ ]]; then
+      echo "pool ${NTP_SERVER} iburst" >> /var/run/chrony/chrony.conf
+  # In all other cases, we should add dot, then pass it to file
+  else
+      echo "pool ${NTP_SERVER}. iburst" >> /var/run/chrony/chrony.conf
+  fi
```
Comment (RomanenkoDenys) [inline modules/470-chrony/images/chrony/start.sh:50]: Why we do not do it in the helm manifest ?

**Review by RomanenkoDenys — COMMENTED** [review]
question

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Pull request merged**